scripts/clean-shrinkwrap.js

```diff
@@ -7,264 +7,101 @@
-        return _.every([
-          _.find(blacklist, {
-            name,
-            version: dependency.version
-          }),
-          dependency.dev,
-          dependency.optional
-        ])
-      })
-      .mapValues((dependency) => {
-        return removeOptionalDevelopmentDependencies(dependency, blacklist)
-      })
-      .value()
+console.log('Cleaning shrinkwrap...')
+
+// Walk the generated shrinkwrap tree & apply modifications if necessary
+traverseDeps(shrinkwrap, (parent, parentName, name, info) => {
+  // If this dependency depends on a "blacklisted" optional
+  // dependency; remove it from the shrinkwrap
+  if (packageInfo.platformSpecificDependencies.includes(name)) {
+    console.log(' - Removing "%s" from "%s"', name, parentName)
+    parent.dependencies[name] = undefined
+    Reflect.deleteProperty(parent.dependencies, name)
+    return
   }
 
-  return shrinkwrap
-}
-
-/**
- * @summary Get the dependency tree of a dependency plus the dependency itself
- * @function
- * @private
- *
- * @param {Object} shrinkwrap - the shrinkwrap file contents
- * @param {String[]} shrinkwrapPath - path to shrinkwrap dependency
- * @returns {Object[]} tree
- *
- * @example
- * const tree = getTree(require('./npm-shrinkwrap.json'), [ 'drivelist' ]);
- *
- * _.each(tree, (dependency) => {
- *   console.log(dependency.name);
- *   console.log(dependency.path);
- *   console.log(dependency.version);
- * });
- */
-const getTree = (shrinkwrap, shrinkwrapPath) => {
-  return _.compact(_.concat([
-    getPrettyShrinkwrapDependencyObject(shrinkwrap, shrinkwrapPath)
-  ], getDependencyTree(shrinkwrap, shrinkwrapPath)))
-}
-
-const blacklist = _.reduce(platformSpecificDependencies, (accumulator, dependencyPath) => {
-  return _.concat(accumulator, getTree(shrinkwrapFile, dependencyPath))
-}, [])
+  // Ensure the `resolved` field contains a HTTPS registry URL,
+  // as some versions of npm apparently fall back to HTTP on some platforms
+  // under some circumstances
+  if (/^http:\/\//.test(info.resolved)) {
+    info.resolved = info.resolved.replace(/^http/, 'https')
+  }
 
-const filteredShrinkwrap = removeOptionalDevelopmentDependencies(shrinkwrapFile, blacklist)
-const result = JSON.stringify(filteredShrinkwrap, null, JSON_INDENTATION_SPACES)
+  // Delete `from` fields to avoid different diffs
+  // on different platforms
+  info.from = undefined
+  Reflect.deleteProperty(info, 'from')
+})
+
+// Generate the new shrinkwrap JSON
+const shrinkwrapJson = JSON.stringify(shrinkwrap, null, JSON_INDENT)
+
+// Write back the modified npm-shrinkwrap.json
+fs.writeFile(SHRINKWRAP_FILENAME, shrinkwrapJson, (error) => {
+  if (error) {
+    console.log(`[ERROR] Couldn't write shrinkwrap file: ${error.stack}`)
+    process.exit(1)
+  } else {
+    console.log(`[OK] Wrote shrinkwrap file to ${path.relative(__dirname, SHRINKWRAP_FILENAME)}`)
+  }
+})
 
-fs.writeFileSync(NPM_SHRINKWRAP_FILE_PATH, `${result}\n`)
-console.log('Done')
+console.log('')
```

Comment:
Huh?

Reply:
This just emits an empty line, for spacing output to avoid bleeding eyes :)